support/controlplane-component/controlplane-component.go

```diff
@@ -326,8 +337,78 @@ func (c *controlPlaneWorkload[T]) reconcileWorkload(cpContext ControlPlaneContex
 		return err
 	}
 
+	// Ensure the workload has a finalizer for cleanup
```

Comment:
Per discussion with @csrwng during our IBM/Red Hat interlock meeting, I think that we need this scoped to the components of interest since applying the finalizer to all components may break how ROKS handles cluster deletion.

Reply:
Yes, I am currently reworking this to correctly apply to the relevant deployments.